### PR TITLE
UCM/EVENT: fixed ucp_test_event call

### PR DESCRIPTION
**Pull request by hoopoepg**
- there were issue in ucm_event_test call: it could false
  positive detect missing or over-called events
- re-implemented function using event-by-event check

### CODE REVIEW

File: src/ucm/mmap/install.c [hunk 1]
```diff
@@ -104,23 +110,43 @@ void ucm_fire_mmap_events(int events)
...
 ucs_status_t ucm_mmap_test_events(int events, int *out_events)
 {
+    ucm_mmap_test_event_data_t data = {.out_events = out_events};
+    int native_events;
+    int event;
     ucm_event_handler_t handler;
 
+    native_events = (events & UCM_EVENT_VM_MAPPED   ? UCM_NATIVE_EVENT_VM_MAPPED   : 0) |
+                    (events & UCM_EVENT_VM_UNMAPPED ? UCM_NATIVE_EVENT_VM_UNMAPPED : 0) |
+                    (events & ~(UCM_EVENT_VM_MAPPED | UCM_EVENT_VM_UNMAPPED));
```
Comment (yosefe) [inline src/ucm/mmap/install.c]: don't understand this line, we get here only VM_MAPPED and VM_UNMAPPED were not set, so what's the purpose to remove those bits?

Comment (hoopoepg) [reply src/ucm/mmap/install.c]: we are checking for native events only: VM_**** are expanded into set of native events which should be fired, and moreover: **all** native events must be fired for specific aggregate event. fire_event call fires multiple events for every aggregate event which makes impossible to identify missing hooks

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10446/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10462/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10463/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
forced push to resolve conflicts

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10467/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10468/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10471/ for details (Mellanox internal link).

File: src/ucm/mmap/install.c [hunk 2]
```diff
@@ -69,66 +84,160 @@ static void ucm_mmap_event_test_callback(ucm_event_type_t event_type,
...
 {
     int shmid;
     void *p;
 
     if (events & (UCM_EVENT_MMAP|UCM_EVENT_MUNMAP|UCM_EVENT_MREMAP|
                   UCM_EVENT_VM_MAPPED|UCM_EVENT_VM_UNMAPPED)) {
-        p = mmap(NULL, 0, 0, 0, -1 ,0);
-        p = mremap(p, 0, 0, 0);
-        munmap(p, 0);
+        UCM_FIRE_EVENT(events, UCM_EVENT_MMAP|UCM_EVENT_VM_MAPPED,
```
Comment (yosefe) [inline src/ucm/mmap/install.c]: why need to set/unset callback for each test?
can't we just clear the state of "received events" before every test?

Comment (hoopoepg) [reply src/ucm/mmap/install.c]: the problem is in external event handlers: OMPI produces only VM_UNMAP events and we can't know which call produced event without setting personal handler per call

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10488/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
@yosefe anything else to fix?

File: src/ucm/mmap/install.c [hunk 3]
```diff
@@ -62,72 +77,124 @@ static ucm_mmap_func_t ucm_mmap_funcs[] = {
...
+static void ucm_fire_mmap_empty_cb(int event, int pre_call, void *arg)
+{
+}
+
+void ucm_fire_mmap_events(int events)
+{
+    ucm_fire_mmap_events_with_cb(events, ucm_fire_mmap_empty_cb, NULL);
+}
+
+static void ucm_mmap_test_events_cb(int events, int pre_call, void *arg)
```
Comment (yosefe) [inline src/ucm/mmap/install.c]: per phone discussion, let's remove the callback and update the flags directly from the macro

Comment (hoopoepg) [reply src/ucm/mmap/install.c]: done

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10512/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10513/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10514/ for details (Mellanox internal link).

**Comment by brminich** [comment]
bot:mlx:retest

**Comment by brminich** [comment]
bot:bgate:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10519/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
make distcheck fails:
```
22:34:01 Making all in .
22:34:01 make[4]: Entering directory `/scrap/jenkins/workspace/hpc-ucx-pr-4/label/hpc-arm-cavium-jenkins/worker/2/build-test/ucx-1.7.0/_build/sub/src/ucm'
22:34:01   CC       event/libucm_la-event.lo
22:34:01   CC       malloc/libucm_la-malloc_hook.lo
22:34:01   CC       mmap/libucm_la-install.lo
22:34:01   CC       util/libucm_la-replace.lo
22:34:01   CC       util/libucm_la-log.lo
22:34:01   CC       util/libucm_la-reloc.lo
22:34:01   CC       util/libucm_la-sys.lo
22:34:01   CC       bistro/libucm_la-bistro.lo
22:34:01   CC       bistro/libucm_la-bistro_x86_64.lo
22:34:01   CC       bistro/libucm_la-bistro_aarch64.lo
22:34:02   CC       bistro/libucm_la-bistro_ppc64.lo
22:34:02   CC       ptmalloc286/libucm_la-malloc.lo
22:34:03 make[4]: *** [mmap/libucm_la-install.lo] Error 1
22:34:03 make[4]: *** Waiting for unfinished jobs....
22:34:10 make[4]: Leaving directory `/scrap/jenkins/workspace/hpc-ucx-pr-4/label/hpc-arm-cavium-jenkins/worker/2/build-test/ucx-1.7.0/_build/sub/src/ucm'
22:34:10 make[3]: *** [all-recursive] Error 1
22:34:10 make[3]: Leaving directory `/scrap/jenkins/workspace/hpc-ucx-pr-4/label/hpc-arm-cavium-jenkins/worker/2/build-test/ucx-1.7.0/_build/sub/src/ucm'
22:34:10 make[2]: *** [all-recursive] Error 1
22:34:10 make[2]: Leaving directory `/scrap/jenkins/workspace/hpc-ucx-pr-4/label/hpc-arm-cavium-jenkins/worker/2/build-test/ucx-1.7.0/_build/sub'
22:34:10 make[1]: *** [all] Error 2
22:34:10 make[1]: Leaving directory `/scrap/jenkins/workspace/hpc-ucx-pr-4/label/hpc-arm-cavium-jenkins/worker/2/build-test/ucx-1.7.0/_build/sub'
```

**Comment by hoopoepg** [comment]
yep, really strange issue...

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10522/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
@yosefe squash changes?

**Comment by hoopoepg** [comment]
forced push to squash & resolve conflicts

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10528/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10531/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by hoopoepg** [comment]
bot:bgate:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10532/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
@yosefe ok to merge?

**Pull request merged**